nixos/modules/services/security/bitwarden-directory-connector.nix

```diff
@@ -0,0 +1,336 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  connector = pkgs.bitwarden-directory-connector;
+
+  cfg = config.services.bitwarden_directory_connector;
+
+  nodejs = pkgs.nodejs-18_x;
+
+  ldap_data = ''
+    {
+      "ssl": ${boolToString cfg.ldap.ssl},
+      "startTls": ${boolToString cfg.ldap.startTls},
+      "sslAllowUnauthorized": ${boolToString cfg.ldap.sslAllowUnauthorized},
+      "port": ${toString cfg.ldap.port},
+      "currentUser": false,
+      "ad": ${boolToString cfg.ldap.ad},
+      "pagedSearch": true,
+      "password": "to_be_replaced",
+      "hostname": "${cfg.ldap.hostname}",
+      "rootPath": "${cfg.ldap.root}",
+      "username": "${cfg.ldap.username}"
+    }
+  '';
+
+  sync_data = ''
+    {
+      "removeDisabled": ${boolToString cfg.sync.removeDisabled},
+      "overwriteExisting": ${boolToString cfg.sync.overwriteExisting},
+      "largeImport": ${boolToString cfg.sync.largeImport},
+      "creationDateAttribute": "${cfg.sync.creationDateAttribute}",
+      "memberAttribute": "${cfg.sync.memberAttribute}",
+
+      "useEmailPrefixSuffix": ${boolToString cfg.sync.emailPrefixSuffix.enable},
+      ${optionalString cfg.sync.emailPrefixSuffix.enable ''
+      "emailPrefixAttribute": "${cfg.sync.emailPrefixSuffix.prefixAttribute}",
+      "emailSuffix": "${cfg.sync.emailPrefixSuffix.suffix}",
+    ''}
+
+      "users": ${boolToString cfg.sync.users.enable},
+      ${optionalString cfg.sync.users.enable ''
+      "userPath": "${cfg.sync.users.path}",
+      "userObjectClass": "${cfg.sync.users.objectClass}",
+      "userEmailAttribute": "${cfg.sync.users.emailAttribute}",
+      "userFilter": "${cfg.sync.users.filter}",
+    ''}
+
+      "groups": ${boolToString cfg.sync.groups.enable},
+      ${optionalString cfg.sync.groups.enable ''
+      "groupPath": "${cfg.sync.groups.path}",
+      "groupObjectClass": "${cfg.sync.groups.objectClass}",
+      "groupNameAttribute": "${cfg.sync.groups.nameAttribute}",
+      "groupFilter": "${cfg.sync.groups.filter}",
+    ''}
+
+      "interval": 5
```

Comment:
Why is this not configurable? Is this every 5 seconds/minutes/hours/..?

Reply:
This is one of the values generated by the the tool itself, it shares its codebase with teh desktop tool.  
Its not used at all in the cli version, which is why its not able to be configured.  
``sync.interval`` is where the user can set the sync interval.